app/src/main/java/com/allan/boardbuddies/views/CanvasView.java

```diff
@@ -65,12 +81,26 @@ public boolean onTouchEvent(MotionEvent event) {
 
         switch (event.getAction()) {
             case MotionEvent.ACTION_DOWN:
-                startDraw(x, y);
+                TextBox foundBox = findPressTextBox(x, y);
+                if (foundBox != null){
+                    selectedBox = foundBox;
+                } else {
+                    startDraw(x, y);
+                }
                 break;
             case MotionEvent.ACTION_MOVE:
-                continueDraw(x, y);
+                if (selectedBox != null){
+                    touchMoveText(x,y);
+                } else {
+                    continueDraw(x, y);
+                }
                 break;
             case MotionEvent.ACTION_UP:
+                if (selectedBox != null){
```

Comment:
might need to rethink this once you start to support edit - maybe you should null out the selectedBox on the next ACTION_DOWN instead

Reply:
we'll get there when we get there